code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -412,8 +412,7 @@
 		playsound(get_turf(user),A.load_audio,50,TRUE)
 		to_chat(user, span_notice("You add [A.rounds] [A.ammo_type][A.rounds > 1?"s":""] to the [gun.name]"))
 		A.rounds = 0
-		A.set_custom_materials(list(/datum/material/iron=2000))
```

Comment:
this is supposed to work in tandem with to dynamically change the material content of the box as it depletes its' ammo count.
https://github.com/tgstation/tgstation/blob/5d175ee2b5027d58f44c6bb6f7876a71c48c908d/code/modules/vehicles/mecha/mecha_defense.dm#L402-L404

but due to ammo boxes not having any `custom_materials` set this never fires. if the values are set, however, the math doesn't account for the box to have at least 2k of iron content. it's supposed to have this much iron at least because it has this amount of iron content when emptied and can be folded into a sheet of iron, which is worth 2k.

Reply:
so the issues are:
1. ammo boxes do not have any material content until they are emptied or flattened into a iron sheet
2. ammo boxes _are supposed to_ have their material content change based on the amount of ammo remaining in the box, but it never changes due to issue 1
3. the math that's _supposed_ to handle issue 2 does not account for consistency in iron content, as the box should have at least 2k worth of iron content. if material cost is restored without fixing the math, it may result in partially spent ammo boxes having less iron than they would have if fully spent

I don't think I should fix these in this PR to avoid clutter though, so I guess this PR is ready for review otherwise